lib/torrent.js

```diff
@@ -473,19 +474,15 @@ class Torrent extends EventEmitter {
 
     this._rarityMap = new RarityMap(this)
 
+    this.files = this.files.map(file => new File(this, file))
+
     let rawStore = this._preloadedStore
     if (!rawStore) {
       rawStore = new this._store(this.pieceLength, {
-        // opts.torrent is deprecated (replaced by the name property).
-        // it doesn't appear to be used by current versions of any stores on npm.
-        torrent: {
-          infoHash: this.infoHash
-        },
-        files: this.files.map(file => ({
-          path: path.join(this.path, file.path),
-          length: file.length,
-          offset: file.offset
-        })),
+        ...this.storeOpts,
+        torrent: this,
+        path: this.path,
+        files: this.files,
```

Comment:
does the store need the whole `File` instance reference?

Reply:
it does simply things nicely, and there really isn't a reason not to do it, additionally custom webtorrent specific stores can for example listen to the 'done' event to save a full file to drive, or close an opened stream to the file

Reply:
The current change, as discussed here https://github.com/feross/fs-chunk-store/pull/17#discussion_r671794460 requires that the infoHash is in the path name.

I realize this is useful to prevent collisions, but is probably not what casual users want, and if we change it that would have to be in a major version of webtorrent (2.0).

I think we need to think through the logic a bit more carefully. My current rough idea:
* Don't change the store at all, at least to solve this specific issue
* Add a new param to the torrent opts: `basePath`, which you set instead of `path` if you want the infoHash appended to avoid conflicts. If you set the normal `torrentOpts.path` option instead the behavior won't change.
* In `_onParsedTorrent`, do something like this:
```js
if (!this.path) {
  this.path = path.join(this.basePath || TMP, this.infoHash)
  this.cleanupPath = this.path
}
```

* In `torrent._destroy`, if `torrent.cleanupPath` is set, recursively delete it here in webtorrent, and set `torrent._destroyStoreOnDestroy` to false to disable the store's cleanup logic.

Reply:
I really don't think that webtorrent should have any involvement in the actual deletion of files tho